packages/ia-components/sandbox/details/AnchorDownload.jsx

```diff
@@ -78,36 +78,39 @@ function downloadViaAnchor(el, source, options, cb) {
 }
 
 function reachable({ disconnected, source, identifier, filename }) {
+  // Note that multi element arrays arent reachable if disconnected because they need "compress" which we dont currently support on dweb-mirror
   return (!disconnected
     || (source
-      ? (Array.isArray(source) ? ((source.length === 1) && source[0].downloaded) : source.downloaded )
+      ? (Array.isArray(source) ? source.every(s => s.downloaded) : source.downloaded )
       : (identifier && !filename) // Just an identifier = want directory
     ));
 }
 class AnchorDownload extends React.Component {
   constructor(props) {
+    // TODO-NAMING check that these urls get routed - if just anchors then have problems on dweb-mirror
     super(props);
     this.onClick = this.onClick.bind(this);
     // TODO-STATE this might have the issue of constructor not being re-run and needing componentDidMount catch
     // this.props passes identifier which is required for Dweb, but typically also passes tabIndex, class, title
-    this.state = {
-      url: new URL((this.props.filename && typeof this.props.filename === 'string') // filename = foo.jpg
-        ? `https://archive.org/download/${this.props.identifier}/${this.props.filename}`
-        : (this.props.source && Array.isArray(this.props.source) && this.props.source.length === 1) // source = [ArchiveFile]
+    const url = new URL((this.props.filename && typeof this.props.filename === 'string') // filename = foo.jpg
```

Comment:
in this file (and others), can we break all these nested ternaries down for better readability/scanning?

Reply:
I'd hate to do that, nested ternaries can be hard to read if used in a fully-freeform manner, but I believe there is a fundamental pattern here of 
```
test1
? result1
: test2 
? result2
: result-else
```
Which is not easily expressed in Javascript in other ways, certainly not more readably.
Which is why I line them up one above the other (though lint tends to rearrange them).